Makefile

```diff
@@ -145,39 +156,59 @@ run-% : $(OBJDIR)/%
 	@tests/tap.sh $(<:build/%=%)
 
 test : $(tests:$(OBJDIR)/%=run-%) $(examples:$(OBJDIR)/%=run-%)
+# run test target in parallel
+tst : ;@$(MAKE) $(MFLAGS) V=$(V) test
+# CPU C tests only for backend %
+ctc-% : $(ctests);@$(foreach tst,$(ctests),$(tst) /cpu/$*;)
 
 prove : $(tests) $(examples)
 	$(PROVE) $(PROVE_OPTS) --exec 'tests/tap.sh' $(tests:$(OBJDIR)/%=%) $(examples:$(OBJDIR)/%=%)
+# run prove target in parallel
+prv : ;@$(MAKE) $(MFLAGS) V=$(V) prove
 
 examples : $(examples)
 
 $(ceed.pc) : pkgconfig-prefix = $(abspath .)
 $(OBJDIR)/ceed.pc : pkgconfig-prefix = $(prefix)
-.INTERMEDIATE: $(OBJDIR)/ceed.pc
+.INTERMEDIATE : $(OBJDIR)/ceed.pc
 %/ceed.pc : ceed.pc.template | $$(@D)/.DIR
 	@sed "s:%prefix%:$(pkgconfig-prefix):" $< > $@
 
+# OCCA env, OKL files cache, clear & rule to install
+OCCA_CACHE_DIR     ?= ${HOME}/.occa
+OCCA_LIB_CACHE_DIR := $(OCCA_CACHE_DIR)/libraries
+OKL_KERNELS        := $(shell find backends/occa -type f -name '*.okl')
+OKL_CACHED_KERNELS := $(subst backends/occa,$(OCCA_LIB_CACHE_DIR)/ceed,$(OKL_KERNELS))
+okl-cache: $(OKL_CACHED_KERNELS)
+okl-clear: ;@occa clear -y -l ceed
+$(OCCA_LIB_CACHE_DIR)/ceed/%.okl: backends/occa/*.okl
+	echo find . -type f -name $*.okl
+	@echo "Caching: $(shell find . -type f -name $*.okl)"
+	@occa cache ceed $(shell find . -type f -name $*.okl)
```

Comment:
So if I install, delete ~/.occa, and then try to run, I get the following error, even if I go back and rebuild the cache. How is this supposed to work?

```
$ ./bp1-petsc -ceed_resource /cpu/occa                                                                                                  
Process decomposition: 1 1 1                                                                                                                                                                  
Local elements: 1000 = 10 10 10
Owned dofs: 1331 = 11 11 11
CEED-OCCA error @ /home/jed/src/libceed/backends/occa/ceed-occa-qfunction.c:198 CeedQFunctionCreate_Occa
Can not find OKL file!
```

Reply:
The user needs to provide at least the OKL file for his qfunctions, I can push the one for petsc

Reply:
I pushed the OKL file for the PETSc example